src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs

```diff
@@ -0,0 +1,270 @@
+			set { SetValue(PosterSourceProperty, value); }
+		}
+
+		public static DependencyProperty PosterSourceProperty { get; } =
+			DependencyProperty.Register(
+				nameof(PosterSource),
+				typeof(ImageSource),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(default(ImageSource), OnPosterSourceChanged));
+
+		private static void OnPosterSourceChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+		{
+			sender.Maybe<MediaPlayerElement>(mpe =>
+			{
+				var source = args.NewValue as ImageSource;
+
+				if (source != null && (mpe.MediaPlayer == null || mpe.MediaPlayer.PlaybackSession.PlaybackState == MediaPlaybackState.None) && mpe._posterImage != null)
+				{
+					mpe._posterImage.Visibility = Visibility.Visible;
+					mpe._mediaPlayerPresenter.Opacity = 0;
+				}
+			});
+		}
+
+		#endregion
+
+		#region AutoPlay Property
+
+		public bool AutoPlay
+		{
+			get { return (bool)GetValue(AutoPlayProperty); }
+			set { SetValue(AutoPlayProperty, value); }
+		}
+
+		public static DependencyProperty AutoPlayProperty { get; } =
+			DependencyProperty.Register(
+				nameof(AutoPlay),
+				typeof(bool),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(true, OnAutoPlayChanged));
+
+		private static void OnAutoPlayChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+		{
+			sender.Maybe<MediaPlayerElement>(mpe =>
+			{
+				if (mpe.MediaPlayer != null)
+				{
+					mpe.MediaPlayer.AutoPlay = (bool)args.NewValue;
+				}
+			});
+		}
+
+		#endregion
+
+		#region MediaPlayer Property
+
+		public Windows.Media.Playback.MediaPlayer MediaPlayer
+		{
+			get { return (Windows.Media.Playback.MediaPlayer)GetValue(MediaPlayerProperty); }
+			set { SetValue(MediaPlayerProperty, value); }
+		}
+
+		public static DependencyProperty MediaPlayerProperty { get; } =
+			DependencyProperty.Register(
+				nameof(MediaPlayer),
+				typeof(Windows.Media.Playback.MediaPlayer),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(default(Windows.Media.Playback.MediaPlayer), OnMediaPlayerChanged));
+
+		private static void OnMediaPlayerChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+		{
+			sender.Maybe<MediaPlayerElement>(mpe =>
+			{
+				if (args.OldValue is Windows.Media.Playback.MediaPlayer oldMediaPlayer)
+				{
+					oldMediaPlayer.MediaFailed -= mpe.OnMediaFailed;
+					oldMediaPlayer.MediaFailed -= mpe.OnMediaOpened;
+				}
+
+				if (args.NewValue is Windows.Media.Playback.MediaPlayer newMediaPlayer)
+				{
+					newMediaPlayer.Source = mpe.Source;
+					newMediaPlayer.MediaFailed += mpe.OnMediaFailed;
+					newMediaPlayer.MediaOpened += mpe.OnMediaOpened;
+					mpe.TransportControls?.SetMediaPlayer(newMediaPlayer);
+					mpe._isTransportControlsBound = true;
+				}
+			});
+		}
+
+		private void OnMediaFailed(Windows.Media.Playback.MediaPlayer session, object args)
+		{
+			Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
```

Comment:
Why is this (and below) dispatched?

Reply:
Dispatched because it's UI update made from player background thread